app/controllers/concerns/application.rb

```diff
@@ -0,0 +1,21 @@
+module Application
+  extend ActiveSupport::Concern
+
+  included do
+    helper_method :current_order
+  end
+
+  def current_user
+    super&.decorate
+  end
+
+  def set_order
+    session[:order_id] = Order.find_or_create_by(id: session[:order_id]).id
+    session[:order_items_count] = Order.find(session[:order_id]).order_items.count
+  end
+
+  def current_order
+    set_order
+    Order.find(session[:order_id]).decorate
+  end
```

Comment:
Can we replace these methods to 

```
def current_order
  Order.find_or_create_by(id: session[:order_id]).decorate
end
```
and use `current_order.order_items.count` if we need counts of order_items in the view?